### PR TITLE
Add control system measurement initialization

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/ControlSystem/Actions/Test_InitializeMeasurements.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  using submeasurements = tmpl::list<Submeasurement<ExpectedSystems>>;
+};
+
+struct SystemB;
+
+struct SystemA : tt::ConformsTo<control_system::protocols::ControlSystem> {
+  static std::string name() { return "A"; }
+  using measurement = Measurement<tmpl::list<SystemA, SystemB>>;
+  struct process_measurement {};
+};
```
Comment (knelli2) [inline tests/Unit/ControlSystem/Actions/Test_InitializeMeasurements.cpp]: I'm surprised this conforms to the protocol. Doesn't it also need a `using simple_tags = tmpl::list<>`?
Same for the others in this PR.

File: src/ControlSystem/Metafunctions.hpp [hunk 2]
```diff
@@ -44,5 +44,45 @@ using control_systems_with_measurement_t =
...
+template <typename Submeasurement, typename ControlSystems>
+struct interpolation_target_tags_for_submeasurement {
+ private:
+  using declared_type =
+      typename Submeasurement::template interpolation_target_tag<
+          ControlSystems>;
+
+ public:
+  using type = tmpl::conditional_t<std::is_same_v<declared_type, void>,
+                                   tmpl::list<>, tmpl::list<declared_type>>;
```
Comment (knelli2) [inline src/ControlSystem/Metafunctions.hpp:70]: Just a question. Why does the conforming type have to specify void if it's just going to end up as an empty list anyways? Why can't it just originally be an empty list?

File: tests/Unit/ControlSystem/Actions/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+set(LIBRARY_SOURCES
+  ${LIBRARY_SOURCES}
+  Actions/Test_InitializeMeasurements.cpp
```
Comment (knelli2) [inline tests/Unit/ControlSystem/Actions/CMakeLists.txt]: Could you also move the `Test_Initialization.cpp` into the `Actions` namespace in a separate commit. We should probably keep things consist.

**Review by knelli2 — CHANGES_REQUESTED** [review]

File: src/ControlSystem/Metafunctions.hpp (hunk 2 above)
Comment (wthrowe) [reply src/ControlSystem/Metafunctions.hpp:70]: The alias isn't a list, just a type.

File: tests/Unit/ControlSystem/Actions/Test_InitializeMeasurements.cpp (hunk 1 above)
Comment (wthrowe) [reply tests/Unit/ControlSystem/Actions/Test_InitializeMeasurements.cpp]: Probably nothing used here does a check.  I've added some assertions.

**Review by knelli2 — DISMISSED** [review]
Looks good. You can squash.

**Review by knelli2 — DISMISSED** [review]

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
@wthrowe this needs a rebase

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
ignoring failure of test to compile on (gcc-10, Debug, ON)

**Pull request merged**